shm/src/Ship.cpp

```diff
@@ -42,3 +44,44 @@ std::shared_ptr<Cargo> Ship::getCargo(size_t index) const {
     }
     return cargo_[index];
 }
+
+void Ship::load(const std::shared_ptr<Cargo>cargo) {
+    if (cargo != nullptr) {
+        bool ifExistCargo = false;
+        for (auto const & element : cargo_) {
+            if (element == cargo) {
+                *element+=cargo->getAmount();
+                ifExistCargo = true;
+                break;
+            }
+        }
+        if (ifExistCargo == false) {
+            cargo_.push_back(cargo);
+        }
+    }
+}
+
+void Ship::unload(Cargo* cargo) {
+    if (cargo != nullptr) {
+        int it = -1;
+        for (int i = 0; i < static_cast<int>(cargo_.size()); i++) {
+            if (*(cargo_[i].get()) == *cargo) {
+                it = i;
+                break;
+            }
+        }
+        if (it == -1) {
+            std::cerr << "Cargo not exist in ship\n";
+        } else if (cargo->getAmount() > cargo[it].getAmount()) {
+            std::cerr << "Wrong value, you wanna unload more then exist in ship\n";
+        } else if (cargo->getAmount() == cargo[it].getAmount()) {
+            cargo_.erase(std::remove(cargo_.begin(), cargo_.end(), cargo_[it]), cargo_.end());
+        } else {
+            *cargo_[it]-= cargo->getAmount();
+        }
+    }
+}
```

Comment:
Problems in lines:
Ship.cpp:68:26
Ship.cpp:78:75
Ship.cpp:80:21

error: implicit conversion changes signedness: 'int' to 'std::__1::vector<std::__1::shared_ptr<Cargo>>::size_type' (aka 'unsigned long')